amgut/lib/data_access/ag_data_access.py

```diff
@@ -357,16 +367,18 @@ def deleteAGParticipantSurvey(self, ag_login_id, participant_name):
                      WHERE ag_login_id = %s AND participant_name = %s"""
             TRN.add(sql, [ag_login_id, participant_name])
 
-            sql = """INSERT INTO ag.consent_revoked
-                     (ag_login_id,participant_name, participant_email)
-                     VALUES (%s, %s, %s)"""
-            sql_args = [[ag_login_id, participant_name, pemail]
-                        for pemail in participant_emails]
-            TRN.add(sql, sql_args, many=True)
+            # only removes user from ag.consent_revoked if already there
+            if ag_login_id not in revoked:
```

Comment:
should this check be done before the delete on line 359? eg:

```python
if ag_login_id in revoked:
    return  # no need to do anything
```

Reply:
I put it in after these two statements

 363             sql = "DELETE FROM ag_login_surveys WHERE survey_id IN %s"
 364             TRN.add(sql, [tuple(survey_ids)])
 365 
 366             sql = """DELETE FROM ag_consent
 367                      WHERE ag_login_id = %s AND participant_name = %s"""
 368             TRN.add(sql, [ag_login_id, participant_name])

because I assumed they might still need to be called even if the id was still in consent_revoked. I figured they would cause an error if the revoked check needed to be put ahead of them.